### PR TITLE
Fix order of reading notifications for tor launch.

### PR DESCRIPTION
**Pull request by riastradh-brave**
Also fix the polling mechanism, and tidy up after ourselves.

The last commit on this branch adds a bunch of diagnostics that turned out not to be very helpful.  Maybe they would be helpful in the future, so maybe we should have an option to turn them on, but I'm inclined to just remove them because they weren't useful for much of anything besides initial development.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [x] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
- #14517
- `npm run unittest`

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by diracdeltas** [comment]
can we remove some/all of the console.logs?

**Comment by riastradh-brave** [comment]
@diracdeltas Yes.  We can excise the last commit (the WIP one) altogether.

File: app/tor.js [hunk 1]
```diff
@@ -288,7 +293,7 @@ class TorDaemon extends EventEmitter {
    */
   _polled () {
     assert(this._polling)
-    if (this._retry_polling && this._control === null) {
+    if (this._retry_poll && this._control === null) {
```
Comment (diracdeltas) [inline app/tor.js]: ++ haha

**Comment by riastradh-brave** [comment]
I removed the commit with the logs.

**Pull request merged**